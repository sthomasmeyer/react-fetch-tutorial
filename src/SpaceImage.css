body {
  background-color: #00040d;
}

.SpaceImage {
  background-color: ghostwhite;
  margin: auto;
  width: 80%;
  border: 8px solid lightcoral;
  border-radius: 7%;
  padding: 21px;
  align-items: center;
  text-align: center;
  overflow: auto;
}

.SpaceImage-header {
  color: #00040d;
  font-size: 2rem;
  font-family: 'Noto Sans', sans-serif;
}

.SpaceImage-image {
  max-height: 100%;
  max-width: 100%;
}
