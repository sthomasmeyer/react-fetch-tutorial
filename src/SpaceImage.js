// Import React itself (+) the 'useState' and 'useEffect' hooks.
import React, { useState, useEffect } from 'react';
import './SpaceImage.css';
import { NASA_API_KEY } from './secrets';

function SpaceImage() {
  // Declare three new state variables --> 'error', 'isLoaded', and 'image'
  const [error, setError] = useState(null);
  const [isLoaded, setIsLoaded] = useState(false);
  const [image, setImage] = useState([]);

  useEffect(() => {
    // After the first query parameter 'api_key' (prefixed by a [?])...
    // is added to the url, subsuquent query parameters are preceded by...
    // an ampersand [&].
    fetch(
      `https://api.nasa.gov/planetary/apod?api_key=${NASA_API_KEY}&date=2018-08-11`
    )
      .then((res) => res.json())
      .then(
        (result) => {
          setIsLoaded(true);
          setImage(result);
          console.log(result);
        },
        (error) => {
          setIsLoaded(true);
          setError(error);
        }
      );
  }, []);

  if (error) {
    return <div>Error: {error.message}</div>;
  } else if (!isLoaded) {
    return <div>Loading...</div>;
  } else {
    return (
      <div className='SpaceImage'>
        <h1 className='SpaceImage-header'>
          The NASA Astronomy Picture of the Day
        </h1>
        <img className='SpaceImage-image' src={image.url} alt='NASA PoD' />
        <p className='SpaceImage-caption'>{image.explanation}</p>
      </div>
    );
  }
}

export default SpaceImage;
