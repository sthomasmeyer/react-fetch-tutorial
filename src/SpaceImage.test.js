import React from 'react';
import { render } from '@testing-library/react';
import SpaceImage from './SpaceImage';

// Smoke test:
it(`renders without crashing`, function () {
  render(<SpaceImage />);
});

// Snapshot test:
it(`renders with the anticipated HTML structure`, function () {
  const { asFragment } = render(<SpaceImage />);
  expect(asFragment()).toMatchSnapshot();
});
