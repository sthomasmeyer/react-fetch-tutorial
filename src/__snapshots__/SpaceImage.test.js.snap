// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with the anticipated HTML structure 1`] = `
<DocumentFragment>
  <div>
    Loading...
  </div>
</DocumentFragment>
`;
